docs/data-ai/data/data-pipelines.md

```diff
@@ -0,0 +1,756 @@
+---
+linkTitle: "Aggregate data automatically"
+title: "Aggregate data automatically"
+weight: 25
+description: "Make processed data automatically available for faster, simpler queries."
+type: "docs"
+tags: ["data pipelines", "aggregation", "materialized views"]
+icon: true
+images: ["/services/icons/data-capture.svg"]
+viamresources: ["sensor", "data_manager"]
+platformarea: ["data", "cli"]
+date: "2025-07-02"
+---
+
+Data pipelines automatically transform raw sensor readings into summaries and insights at a schedule that you choose.
+Viam stores the output of these pipelines in a cache so that you can access complex aggregation results more efficiently.
+When late-arriving data syncs to Viam, pipelines automatically re-run to keep summaries accurate.
+
+For example, you could use a data pipeline to pre-calculate results like "average temperature per hour".
+If you query that information frequently, this can save significant computational resources.
+
+## Prerequisites
+
+Before creating a data pipeline, you must enable data capture from at least one component and begin syncing data with Viam.
+
+Only users with organization owner permissions can create a data pipeline.
+
+## Pipeline management
+
+### Create a pipeline
+
+To define a data pipeline, specify a name, organization, schedule, data source type, and query:
+
+{{< tabs >}}
+{{% tab name="CLI" %}}
+
+Use [`datapipelines create`](/dev/tools/cli/#datapipelines) to create your pipeline:
+
+```sh {class="command-line" data-prompt="$" class="command-line" data-continuation-prompt="2-5"}
+viam datapipelines create \
+  --org-id=<org-id> \
+  --name=sensor-counts \
+  --schedule="0 * * * *" \
+  --data-source-type="standard" \
+  --mql='[{"$match": {"component_name": "sensor"}}, {"$group": {"_id": "$location_id", "avg": {"$avg": "$data.readings.value"}}}]'
+```
+
+To pass your query from a file instead of from inline MQL, pass the `--mql-path` flag instead of `--mql`.
+To query data from the [hot data store](/data-ai/data/hot-data-store/), specify `--data-source-type hotstorage`.
```

Comment:
Are we going to define this for the various language examples below?

Reply:
Added, good call.